src/utils.py

```diff
@@ -1440,6 +1440,7 @@ def run_hill_climbing(
         check_goal: Callable[[_S], bool],
         get_successors: Callable[[_S], Iterator[Tuple[_A, _S, float]]],
         heuristic: Callable[[_S], float],
+        early_termination_heuristic_thresh: Optional[float] = None,
```

Comment:
please add a description of `early_termination_heuristic_thresh` in the docstring (Lines 1446-1454)